### PR TITLE
PT-2018: implement Percona-Toolkit package testing with other products

### PR DESCRIPTION
**Pull request by eleo007**
Implement package testing of Percona-Toolkit (pt). Tests pt installation from [pt, tools, original] repos, pt upgrade and pt installation with the other products: Percona-Server-57 (ps57), Percona-Server-80 (ps80), Percona-XtraDB-Cluster-57 (pxc57), Percona-XtraDB-Cluster-80 (pxc80), MysqL server 57 (upstream57), MySQL server 80 (upstream80), Percona Server for MongoDB 44 (psmdb44), Percona Server for MongoDB 50 (psmdb50) .

The tests run will be initiated by the Jenkins task (pt-package-testing-all and pt-package-testing).

**The install test (pt.yml playbook) will:**
- install pt from **tools, ps, original repos** one after another . Choice of main/experimental/testing repos is made based on install_repo variable passed from the Jenkins task;
- check pt version.

**The upgrade test (pt_upgrade.yml playbook) will:**
- install pt from **tools main repo**;
- enable **tools testing/experimental repo** depending on install_repo variable passed from the Jenkins task;
- upgrade the pt;
- check pt version.

**The "combined" test (pt_with_products.yml playbook) will:**
- install from the **main product repo** and start ps/pxc/psmdb/mysql server (for pxc 1 node will be bootstrapped). Products to install are selected based on  install_with variable passed from the Jenkins task. It is possible to skip selected products test;
- install pt from the **tools repo**. Choice of main/experimental/testing repos is made based on install_repo variable passed from the Jenkins task;
- check pt version.
- run pt-mongodb-summary/pt-mysql-summary tool.
   **Note**: there is no check for the ps/pxc/psmdb/upstream version. The latest available version of the product for specific 
    distribution will be installed.

The test Jenkins tasks and runs with logs are available at:
* https://pxc.cd.percona.com/job/pt-package-testing-all/
* https://pxc.cd.percona.com/job/pt-package-testing/
* https://pxc.cd.percona.com/job/pt-package-testing-all/43/

### CODE REVIEW

**Review by kaushikpuneet07 — APPROVED** [review]
LGTM

**Review by hrvojem — APPROVED** [review]
LGTM

File: playbooks/pt_upgrade.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  - name: include tasks for pt installation
+    include_tasks: ../tasks/install_pt.yml
+
+# will fail till the PT-1959 is fixed
+  - name: check that pt version is correct
+    command: /package-testing/version_check.sh pt
+
+  - name: remove pt packages
+    include: ../tasks/remove_pt.yml
```
Comment (puneet0191) [inline playbooks/pt_upgrade.yml:44]: add an empty line

File: playbooks/pt_upgrade.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    vars:
+      - repo_name: tools
+    when: lookup('env', 'install_repo') == "experimental"
+
+  - name: include tasks for pt installation
+    include_tasks: ../tasks/install_pt.yml
+
+# will fail till the PT-1959 is fixed
+  - name: check that pt version is correct
+    command: /package-testing/version_check.sh pt
```
Comment (puneet0191) [inline playbooks/pt_upgrade.yml:41]: The only problem I face with this approach, is I can never check my changes from custom branches on pipelines, not sure how you manage to run & test your changes, do you provide a custom branch when you run wget?

File: playbooks/pt_with_products.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  - name: run pt-mongodb-summary
+    command: pt-mongodb-summary
+    when: lookup('env', 'install_with') is match('psmdb')
+
+  - name: run pt-mysql-summary
+    command: pt-mysql-summary
+    when: lookup('env', 'install_with') is match('(pxc|ps|upstream)(57|80)$')
+
+  - name: remove pt packages
+    include: ../tasks/remove_pt.yml
```
Comment (puneet0191) [inline playbooks/pt_with_products.yml:85]: same here

File: tasks/enable_only_testing_repo_for_pt.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    apt:
+#     deb: https://repo.percona.com/apt/pool/testing/p/percona-release/percona-release_1.0-21.generic_all.deb
+      deb: https://repo.percona.com/apt/percona-release_latest.generic_all.deb
+    when: ansible_os_family == "Debian"
+
+  - name: add the Percona testing yum repos for CentOS
+    yum: name=https://repo.percona.com/yum/percona-release-latest.noarch.rpm
+    when: ansible_os_family == "RedHat"
+
+  - name: enable {{repo_name}} testing repo
```
Comment (puneet0191) [inline tasks/enable_only_testing_repo_for_pt.yml:14]: you could simply have one single task with enables only a specific repo (set as a variable), less code to deal with, just a suggestion.

File: tasks/install_and_run_ps80.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    when: ansible_os_family == "Debian"
+
+  - name: start mysql service on CentOS
+    systemd:
+      name: mysql
+      state: started
+    when: ansible_os_family == "RedHat"
+
+# On Centos ps is started with random root password. Change ps root password and create cnf with new client password on centos.
+  - name: set root password on CentOS
```
Comment (puneet0191) [inline tasks/install_and_run_ps80.yml:24]: Thanks a lot for this, I would rather use these in pmm2-client package testing, I was handling this part by myself, btw what's the password for ps 8.x on Debian?

File: tasks/install_and_run_psmdb50.yml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# This task enables Percona Server for MongoDB 50 (psmdb50) main repository, installs and starts modgodb.
+
+# Enable psmdb repo and install psmdb
+  - name: include tasks for enabling psmdb 50 repo
+    include_tasks: ../tasks/enable_psmdb_repo.yml
+    vars:
+      psmdb_repo: "psmdb-50 release"
```
Comment (puneet0191) [inline tasks/install_and_run_psmdb50.yml:7]: the same here, I could just have install_and_run_psmdb.yml, pass the psmdb_repo as variable, one file to rule them all

**Review by puneet0191 — APPROVED** [review]
what a wonderful work, just some minor comments, thank you for adding upstream mysql install and run, this is good for PMM team too

File: tasks/install_and_run_ps80.yml (hunk 5 above)
Comment (eleo007) [reply tasks/install_and_run_ps80.yml:24]: discussed in Slack

File: playbooks/pt_upgrade.yml (hunk 2 above)
Comment (eleo007) [reply playbooks/pt_upgrade.yml]: Updated playbooks/tasks to use the package-testing folder cloned by Jenkins ( `../version_check.sh pt ` )

File: tasks/enable_only_testing_repo_for_pt.yml (hunk 4 above)
Comment (eleo007) [reply tasks/enable_only_testing_repo_for_pt.yml]: The same: updated to have `../tasks/enable_only_repo_for_pt.yml` that uses `{{repo_name}} {{ repo_type}}` variables

**Review by puneet0191 — APPROVED** [review]
thanks a lot for this great work!!!

**Pull request merged**